api/src/main/java/org/openmrs/module/referencedemodata/obs/DemoObsGenerator.java

```diff
@@ -73,22 +86,39 @@ public DemoObsGenerator(DemoDataConceptCache conceptCache) {
 	
 	public void createDemoVitalsObs(Patient patient, Encounter encounter, Location location) {
 		for (NumericObsValueDescriptor vitalsDescriptor : getVitalsDescriptors()) {
-			createNumericObsFromDescriptor(vitalsDescriptor, patient, encounter, location);
+			createNumericObsFromDescriptor(vitalsDescriptor, patient, encounter, encounter.getEncounterDatetime(), location);
 		}
 	}
 	
 	public void createDemoLabObs(Patient patient, Encounter encounter, Location location) {
-		for (NumericObsValueDescriptor labDescriptor : getLabDescriptors()) {
-			createNumericObsFromDescriptor(labDescriptor, patient, encounter, location);
+		List<Concept> labConcepts = new ArrayList<>();
+		// Add panels thrice to increase chance of being picked
+		labConcepts.addAll(conceptCache.getConceptsByClass("LabSet"));
+		labConcepts.addAll(conceptCache.getConceptsByClass("LabSet"));
+		labConcepts.addAll(conceptCache.getConceptsByClass("LabSet"));
+		labConcepts.addAll(conceptCache.getConceptsByClass("Test"));
+		
+		Concept c = (Concept) randomArrayEntry(labConcepts.toArray());
+		if (c.getConceptClass() != null && "LabSet".equalsIgnoreCase(c.getConceptClass().getName()) && c.getSet()) {
+			List<Obs> obsPanel = new ArrayList<>(c.getSetMembers().size());
+			for (Concept member : c.getSetMembers()) {
+				obsPanel.add(createDemoLabObs(patient, null, encounter.getEncounterDatetime(), location, member));
+			}
+			createObsGroup(c.getUuid(), patient, encounter, encounter.getEncounterDatetime(), location, obsPanel);
+		} else {
+			createDemoLabObs(patient, encounter, encounter.getEncounterDatetime(), location, c);
 		}
+		
 	}
 	
 	public void createTextObs(String conceptDescriptor, String text, Patient patient, Encounter encounter,
 			Date encounterTime,
 			Location location) {
 		Obs obs = createBasicObs(conceptDescriptor, patient, encounterTime, location);
 		obs.setValueText(text);
-		encounter.addObs(obs);
+		if (encounter != null) {
+			encounter.addObs(obs);
+		}
```

Comment:
When is this the case?

Reply:
This is rather not the case for member panel result obs that shouldn't be directly associated with the encounter but the grouping obs instead.